sycl/test/basic_tests/parallel_for_range.cpp

```diff
@@ -0,0 +1,93 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+int main() {
+  auto asyncHandler = [](exception_list es) {
+    for (auto& e : es) {
+      std::rethrow_exception(e);
+    }
+  };
+
+  // parallel_for, 100 global, 3 local -> fail.
+  try {
+    queue q(asyncHandler);
+    q.submit([&](handler &cgh) {
+        cgh.parallel_for<class a>(nd_range<1>(range<1>(100), range<1>(3)),
+            [=](nd_item<1> id) {});
+    });
+    q.wait_and_throw();
+    assert(false && "Should have thrown exception");
```

Comment:
Precise which one.

Reply:
I originally checked for `nd_range_error` specifically, but we only get that exception if we run on host. If we use the OpenCL calls, we get a more generic `runtime_error` instead.

Reply:
That is curious to have a different exception according to the device...

Reply:
My recollection may have been on an older version of the patch, where all the cases here threw `nd_range_error`. The OpenCL invocation code will call `runtime_error` on any OpenCL error. The two errors that are for running the host kernel (the first two chunks in this patch) are indeed using `runtime_error`, while the code that uses `nd_range_error` is run on all kernels or host invocation before even submitting to the queue.

In terms of test code, this means that all except the last two exceptions end up with `nd_range_error`, and the last two are a `runtime_error`. I can change the other catches to specifically check for `nd_range_error` if that would be clearer.

Reply:
Since you have done all the hard work already, it would be nice to test for the exact required exception according to the specification.